src/components/Tabs/Tab.js

```diff
@@ -72,43 +67,52 @@ const TabContainer = styled.label`
 
 export default class Tab extends React.Component {
 
-  handleClick = (e) => {
-    this.props.handleSelect(this.props.namespace, this.props.index);
+  static propTypes = {
+    /**
+     * Mark tab as selected
+     */
+    active: PropTypes.bool,
+    /**
+     * Vertcal layout flag
+     */
+    vertical: PropTypes.bool,
+    /**
+     * Called when user clicks tab
+     */
+    onTabSelected: PropTypes.func,
+    /**
+     * Disable the tab
+     */
+    disabled: PropTypes.bool,
+    /**
+     * Children within tab
+     */
+    children: PropTypes.node,
+    /**
+     * Tab's index started from 0
+     */
+    index: PropTypes.number,
+  };
 
-    if (this.props.onClick) {
-      this.props.onClick(e);
-    }
+  static defaultProps = {
+    disabled: false,
+    active: false,
   };
 
+  createTabClickHandler = (index) => () => this.props.onTabSelected(index);
```

Comment:
Getting there, but one last step could make this component just a simple styled-components export... You can move this handler creator out into the `renderTabs` loop in `Tabs` and pass it directly to the `onClick` handler on `Tab`. If you do that, you don't need to define a component at all in this file, just export the one created by `styled.div`.

The reason I'm pushing this is that one of the big objectives of this library is to create tools which are as versatile as possible. The more I've expanded and worked on it, the more I find that separating out the visual components from the behavioral ones achieves that goal. In this case, a simple `div` styled like a tab which has no special requirements will be more flexible and useful than one which is coupled to an implementation (that is, one that requires an index to be provided). By moving all of the implementation details into a behavioral component (in this case, `Tabs`), the visual component can be used in a wider variety of circumstances (for instance, if you just wanted to render a visual tab in another component later but didn't need tab-switching functionality).